internal/perms/grants.go

```diff
@@ -343,154 +393,219 @@ func Parse(scopeId, grantString string, opt ...Option) (Grant, error) {
 		}
 	}
 
-	opts := getOpts(opt...)
-
-	// Check for templated values ID, and substitute in with the authenticated
-	// values if so. If we are using a dummy user or account ID, store the
-	// original ID and return it at the end; this is usually the case when
-	// validating grant formats.
-	var origId string
-	if grant.id != "" && strings.HasPrefix(grant.id, "{{") {
-		id := strings.TrimSuffix(strings.TrimPrefix(grant.id, "{{"), "}}")
-		id = strings.TrimSpace(id)
-		switch id {
-		case "user.id", ".User.Id":
-			if opts.withUserId != "" {
-				grant.id = strings.ToValidUTF8(opts.withUserId, string(unicode.ReplacementChar))
-			}
-			// Otherwise, substitute in a dummy value
-			origId = grant.id
-			grant.id = "u_dummy"
-		case "account.id", ".Account.Id":
-			if opts.withAccountId != "" {
-				grant.id = strings.ToValidUTF8(opts.withAccountId, string(unicode.ReplacementChar))
-			}
-			origId = grant.id
-			grant.id = "acctoidc_dummy"
-		default:
-			return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, fmt.Sprintf("unknown template %q in grant %q value", grant.id, "id"))
-		}
+	if grant.id != "" && len(grant.ids) > 0 {
+		return Grant{}, errors.NewDeprecated(errors.InvalidParameter, op, fmt.Sprintf("input grant string %q contains both %q and %q fields", grantString, "id", "ids"))
```

Comment:
Any appetite for adding a ctx parameter to Parse(...), so we can stop using
errors.NewDeprecated in this func?

Reply:
also, can we add consts for `id` and `ids`?

Reply:
I do want to add `ctx` but I don't want to muddle this PR with those changes -- I can do that as a follow-on.

Reply:
Done, sent you a new PR link.